application/controllers/ItemsController.php

```diff
@@ -248,7 +286,26 @@ public function batchEditSaveAction()
             throw new Omeka_Controller_Exception_403;
         }
 
-        if ($itemIds = $this->_getParam('items')) {
+        // Batch all items matching params or only selected items.
+        $editAll = (boolean) $this->_getParam('editAll');
+
+        if ($editAll) {
+            // Get the record ids filtered to Omeka_Db_Table::applySearchFilters().
+            $params = json_decode($this->_getParam('params'), true) ?: array();
+            $alias = $this->_helper->db->getTableAlias();
+            $select = $this->_helper->db
+                ->getSelectForFindBy($params)
+                ->reset(Zend_Db_Select::COLUMNS)
+                ->reset(Zend_Db_Select::LIMIT_COUNT)
```

Comment:
Do you need to reset count and offset? `getSelectForFindBy` shouldn't be applying the pagination in the first place, it's just `findBy` that does that.